ui/src/app/cohort-search/search-bar/search-bar.component.ts

```diff
@@ -1,16 +1,132 @@
+  autocompleteOptions,
+  CohortSearchActions,
+  CohortSearchState,
+  ingredientsForBrand,
+  isAutocompleteLoading,
+} from '../redux';
 
 @Component({
   selector: 'app-search-bar',
   templateUrl: './search-bar.component.html',
   styleUrls: ['./search-bar.component.css']
 })
-export class SearchBarComponent {
+export class SearchBarComponent implements OnInit, OnDestroy {
+  @Input() _type;
   searchTerm = '';
-  constructor(private actions: CohortSearchActions) { }
+  options = [];
+  ingredients: any;
+  loading = false;
+  noResults = false;
+  optionSelected = false;
+  multiIngredient = false;
+  error = false;
+  subscription: Subscription;
+
+  constructor(
+    private ngRedux: NgRedux<CohortSearchState>,
+    private actions: CohortSearchActions
+  ) {}
+
+  ngOnInit() {
+    const errorSub = this.ngRedux
+      .select(autocompleteError())
+      .map(err => !(err === null || err === undefined))
+      .subscribe(err => this.error = err);
+
+    const loadingSub = this.ngRedux
+      .select(isAutocompleteLoading())
+      .subscribe(loading => this.loading = loading);
+
+    const optionsSub = this.ngRedux
+      .select(autocompleteOptions())
+      .subscribe(options => {
+        if (this.searchTerm.length >= 4) {
+          this.options = [];
+          const optionNames = [];
+          options.forEach(option => {
+            if (optionNames.indexOf(option.name) === -1) {
+              optionNames.push(option.name);
+              option.displayName = option.name;
+              const start = option.name.toLowerCase().indexOf(this.searchTerm.toLowerCase());
+              if (start > -1) {
+                const end = start + this.searchTerm.length;
+                option.displayName = option.name.slice(0, start)
+                  + '<span style="color: #659F3D;'
+                  + 'font-weight: bolder;'
+                  + 'background-color: rgba(101,159,61,0.2);'
+                  + 'padding: 2px 0;">'
+                  + option.name.slice(start, end) + '</span>'
+                  + option.name.slice(end);
+              }
+              this.options.push(option);
+            }
+          });
+          this.noResults = this._type === 'drug'
+            && !this.optionSelected
+            && !this.options.length;
+        }
+      });
+
+    const ingredientSub = this.ngRedux
+      .select(ingredientsForBrand())
+      .subscribe(ingredients => {
+        this.ingredients = ingredients;
+        const ingredientList = [];
+        console.log(ingredients);
+        this.ingredients.forEach(item => {
+          ingredientList.push(item.name);
+        });
+        if (ingredientList.length) {
+          this.actions.setCriteriaSearchTerms(ingredientList);
+        }
+        this.multiIngredient = ingredientList.length > 1;
+      });
+
+    this.subscription = errorSub;
+    this.subscription.add(loadingSub);
+    this.subscription.add(optionsSub);
+    this.subscription.add(ingredientSub);
```

Comment:
Is the SearchBarComponent reusable outside of meds search? Seems that this code is closely coupled with meds?